### ISSUE TITLE
Evaluate providing some built-in ORM caching

### ISSUE THREAD

**Issue by Kircheneer** [nautobot/nautobot-plugin-ssot#121]
### Environment
* Nautobot version:  latest
* nautobot-ssot version:  latest

### Proposed Functionality

Have the framework expose something like `get_from_cache(content_type, **kwargs)` with a cache scoped to the execution of a single job.

### Use Case

Objects that are commonly looked up (e.g. `Region`, `Status`) could be looked up from the DB once in the beginning and then, for CRUD operations on models or during loading, just be looked up from memory, speeding up operations as a whole.

**Comment by lampwins** [comment]
Yes, this is very much needed as it is a pattern that is commonly implemented by developers today in ssot jobs.

**Comment by Kircheneer** [comment]
This is implemented as of #296 but needs to be exposed in documentation still.
